app/src/main/java/org/oppia/app/help/faq/FAQFragmentPresenter.kt

```diff
@@ -1,23 +1,113 @@
 ) {
+  private lateinit var binding: FaqFragmentBinding
+  private lateinit var linearLayoutManager: LinearLayoutManager
+  private lateinit var linearSmoothScroller: RecyclerView.SmoothScroller
+
   fun handleCreateView(inflater: LayoutInflater, container: ViewGroup?): View? {
-    val binding = FaqFragmentBinding.inflate(inflater, container, /* attachToRoot= */ false)
-    // NB: Both the view model and lifecycle owner must be set in order to correctly bind LiveData elements to
-    // data-bound view models.
-    binding.lifecycleOwner = fragment
+    val viewModel = getFAQViewModel()
+    linearLayoutManager = LinearLayoutManager(activity.applicationContext)
+    linearSmoothScroller = createSmoothScroller()
+
+    binding = FaqFragmentBinding.inflate(inflater, container, /* attachToRoot = */ false)
+
+    binding.faqFragmentRecyclerView.apply {
+      layoutManager = linearLayoutManager
+      adapter = createRecyclerViewAdapter()
+    }
+
+    binding.let {
+      it.lifecycleOwner = fragment
+      it.viewModel = viewModel
+    }
     return binding.root
   }
+
+  private fun createRecyclerViewAdapter(): BindableAdapter<FAQItemViewModel> {
+    return BindableAdapter.MultiTypeBuilder
+      .newBuilder<FAQItemViewModel, ViewType> { viewModel ->
+        when (viewModel) {
+          is FAQHeaderViewModel -> ViewType.VIEW_TYPE_HEADER
+          is FAQContentViewModel -> ViewType.VIEW_TYPE_CONTENT
+          else -> throw IllegalArgumentException("Encountered unexpected view model: $viewModel")
+        }
+      }
+      .registerViewDataBinder(
+        viewType = ViewType.VIEW_TYPE_HEADER,
+        inflateDataBinding = FaqItemHeaderBinding::inflate,
+        setViewModel = FaqItemHeaderBinding::setViewModel,
+        transformViewModel = { it as FAQHeaderViewModel }
+      )
+      .registerViewDataBinder(
+        viewType = ViewType.VIEW_TYPE_CONTENT,
+        inflateDataBinding = FaqContentBinding::inflate,
+        setViewModel = FaqContentBinding::setViewModel,
+        transformViewModel = { it as FAQContentViewModel }
+      )
+      .build()
+  }
+
+  private fun getFAQViewModel(): FAQViewModel {
+    return viewModelProvider.getForFragment(fragment, FAQViewModel::class.java)
+  }
+
+  private enum class ViewType {
+    VIEW_TYPE_HEADER,
+    VIEW_TYPE_CONTENT
+  }
+
+  private fun createSmoothScroller(): RecyclerView.SmoothScroller {
+    val milliSecondsPerInch = 100f
+
+    return object : LinearSmoothScroller(activity) {
+      override fun getVerticalSnapPreference(): Int {
+        return SNAP_TO_START
+      }
+
+      override fun calculateSpeedPerPixel(displayMetrics: DisplayMetrics?): Float {
+        return milliSecondsPerInch / displayMetrics!!.densityDpi
+      }
+
+      override fun calculateDyToMakeVisible(view: View, snapPreference: Int): Int {
+        return super.calculateDyToMakeVisible(view, snapPreference) + dipToPixels(48)
+      }
+    }
+  }
+
+  private fun dipToPixels(dipValue: Int): Int {
```

Comment:
Do won't need this function once you remove `linearSmoothScroller `

Reply:
Done :)